nautobot/extras/tests/test_views.py

```diff
@@ -1722,6 +1724,144 @@ def setUpTestData(cls):
+        for instance in self._get_queryset().filter(pk__in=pk_list):
+            self.assertEqual(instance.enabled, True)
+            job_class = instance.job_class
+            if job_class is not None:
+                for clear_override_field in clear_override_fields:
+                    overridable_field = clear_override_field[6:-9]
+                    override_field = clear_override_field[6:]
+                    reset_override = self.bulk_edit_data.get(clear_override_field, False)
+                    if overridable_field == "task_queues":
+                        override_value = self.bulk_edit_data.get(overridable_field).split(",")
+                    else:
+                        override_value = self.bulk_edit_data.get(overridable_field)
+                    # if clear_override is true, assert that values are reverted back to default values
+                    if reset_override is True:
+                        self.assertEqual(getattr(instance, overridable_field), getattr(job_class, overridable_field))
+                        self.assertEqual(getattr(instance, override_field), False)
+                    # if clear_override is false, assert that job attribute is set to the new value from the form
+                    elif reset_override is False and (override_value is False or override_value):
+                        self.assertEqual(getattr(instance, overridable_field), override_value)
+                        self.assertEqual(getattr(instance, override_field), True)
+                    # if clear_override is false and no new value is entered, assert that value of the job is unchanged
+                    else:
+                        self.assertEqual(getattr(instance, overridable_field), old_data[instance.pk][overridable_field])
+                        self.assertEqual(getattr(instance, override_field), old_data[instance.pk][overridable_field])
+
+    @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+    def test_bulk_edit_objects_with_permission(self):
+        pk_list = list(self._get_queryset().values_list("pk", flat=True)[:3])
+        instances = self._get_queryset().filter(pk__in=pk_list)
+        overridable_fields = [field for field in JOB_OVERRIDABLE_FIELDS if field != "name"]
+        old_data = {}
+        for instance in instances:
+            old_data[instance.pk] = {}
+            job_class = instance.job_class
+            if job_class is not None:
+                for field in overridable_fields:
+                    old_data[instance.pk][field] = getattr(job_class, field)
+
+        data = {
+            "pk": pk_list,
+            "_apply": True,  # Form button
+        }
+        # Append the form data to the request
+        data.update(post_data(self.bulk_edit_data))
+        # Assign model-level permission
+        self.add_permissions("extras.change_job")
+        # Try POST with model-level permission
+        self.assertHttpStatus(self.client.post(self._get_url("bulk_edit"), data), 302)
```

Comment:
Could this be replaced with a call to `super().test_bulk_edit_objects_with_permission()`?

Reply:
I am not sure if that is possible, since at the end of the generic `test_bulk_edit_objects_with_permission()`, there is a for loop asserting the instance data is the same as the `bulk_edit_data`:
```
for instance in self._get_queryset().filter(pk__in=pk_list):
                self.assertInstanceEqual(instance, self.bulk_edit_data)
```
This for loop is going to raise AssertionErrors because Job's `bulk_edit_data` is different from the generic one with extra `clear_override` fields and values. Same comment below.

Reply:
https://github.com/nautobot/nautobot/pull/5301/commits/79b52fe89a57b01c51b81abe8a84bbf267c6373e